### ISSUE TITLE
Can't bind to 'list' since it isn't a known property of 'input'. (Using html's datalist)

### ISSUE THREAD

**Issue by Judahh** [angular/angular#12759]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Error: Template parse errors:
Can't bind to 'list' since it isn't a known property of 'input'.

**Expected behavior**

**Minimal reproduction of the problem with instructions**
<input 
    *ngIf="modelTextInput.isDataList()"
    name="{{modelTextInput.name}}"
    list="{{modelTextInput.getDataListId()}}"
    placeholder="{{modelTextInput.placeholder}}"
    [style]="style()|safeStyle"
    value="{{modelTextInput.value}}"/>

<datalist 
    *ngIf="modelTextInput.isDataList()"
    id="{{modelTextInput.getDataListId()}}">
    <option 
        *ngFor="let option of modelTextInput.arrayOptions">
        {{option}}
    </option>
</datalist>

**What is the motivation / use case for changing the behavior?**
I want to use html's datalist

**Please tell us about your environment:**
macOS Sierra 10.12.1

* **Angular version:** ~2.0.1

* **Browser:** [Chrome 54.0.2840.71]

 
* **Language:** [TypeScript ^2.0.3 | ES5]

* **Node (for AoT issues):** `node --version` =   7.0.0

**Comment by DzmitryShylovich** [comment]
Hello, we reviewed this issue and determined that it doesn't fall into the bug report or feature request category. This issue tracker is not suitable for support requests, please repost your issue on [StackOverflow](http://stackoverflow.com/) using tag `angular`.

If you are wondering why we don't resolve support issues via the issue tracker, please [check out this explanation](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-got-a-question-or-problem).

**Comment by Judahh** [comment]
What? This isn't a support request.
'list' IS a known property of 'input'.
I need to use html's datalist, and html's input needs to link to datalist's id by the property 'list'.

**Comment by mattdistefano** [comment]
@DzmitryShylovich see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-list

@Judahh In the meantime you can probably use the `[attr.list]=""` syntax.

**Comment by ericmartinezr** [comment]
This is the same case of `for` attribute, see https://github.com/angular/angular/issues/7516. For more info on the syntax https://angular.io/docs/ts/latest/guide/template-syntax.html

**Comment by vicb** [comment]
List is indeed an attribute and not a property.
That's why `[attr.list]=""` should be used here.

**Comment by PeterGursky** [comment]
instead of **list="{{modelTextInput.getDataListId()}}"** use [attr.list]=modelTextInput.getDataListId()

**Comment by LloydTDavies** [comment]
@mattdistefano and @vicb solution of binding to the [attr.list]="" worked wonders for me!
I had built an angular component which is essentially a lookup but I found if I had used this custom component more than once it would share the last updated data list because the list ID's where all the same.

Now I just import a name for that component which is manually set by the developer and make sure the list and input use that.
